File: backend/app/03_high_complexity/src/core/interfaces/types/terrain.py
from typing import Any, Dict, List
from enum import Enum



class TerrainFeatureType(Enum):
    PLAIN = 'PLAIN'
    MOUNTAIN = 'MOUNTAIN'
    WATER = 'WATER'
    FOREST = 'FOREST'
    SWAMP = 'SWAMP'
class TerrainModificationType(Enum):
    LEVEL = 'LEVEL'
    SMOOTH = 'SMOOTH'
    RAMP = 'RAMP'
    FOUNDATION = 'FOUNDATION'
class TerrainModification:
    position: GridPosition
    type: \'TerrainModificationType\'
    params: Dict[str, Any>
class TerrainAnalysisResult:
    slope: float
    buildabilityScore: float
    nearbyFeatures: List[TerrainFeatureType]
    environmentalImpact: float
class TerrainFeature:
    type: \'TerrainFeatureType\'
    position: Coordinates
    size: GridDimensions
class BuildableArea:
    position: Coordinates
    size: GridDimensions
    slope: float
    preferredCategories: List[POICategory]
class TerrainData:
    heightMap: List[List[float]]
    features: List[TerrainFeature]
    buildableAreas: List[BuildableArea] 

File: app/frontend/src/core/interfaces/types/terrain.ts
import { GridDimensions } from './grid';
import { POICategory } from './spatial';
import { Coordinates } from '../poi/types/POITypes';

export enum TerrainFeatureType {
  PLAIN = 'PLAIN',
  MOUNTAIN = 'MOUNTAIN',
  WATER = 'WATER',
  FOREST = 'FOREST',
  SWAMP = 'SWAMP'
}

export enum TerrainModificationType {
  LEVEL = 'LEVEL',
  SMOOTH = 'SMOOTH',
  RAMP = 'RAMP',
  FOUNDATION = 'FOUNDATION'
}

export interface TerrainModification {
  position: GridPosition;
  type: TerrainModificationType;
  params: Record<string, any>;
}

export interface TerrainAnalysisResult {
  slope: number;
  buildabilityScore: number;
  nearbyFeatures: TerrainFeatureType[];
  environmentalImpact: number;
}

export interface TerrainFeature {
  type: TerrainFeatureType;
  position: Coordinates;
  size: GridDimensions;
}

export interface BuildableArea {
  position: Coordinates;
  size: GridDimensions;
  slope: number;
  preferredCategories: POICategory[];
}

export interface TerrainData {
  heightMap: number[][];
  features: TerrainFeature[];
  buildableAreas: BuildableArea[];
} 